### ISSUE TITLE
false positive of lossy_float_literal for bigger powers of 2

### ISSUE THREAD

**Issue by kawogi** [rust-lang/rust-clippy#7404]
Powers of 2 which exceed the size of the mantissa seem to yield a `lossy_float_literal` warning even though those are perfectly representable.

I tried this code:

```rust
fn main() {
    let x64 = 4_294_967_296_f64;
    let x32 = 4_294_967_296_f32;
    println!("x64 {:016x}", x64.to_bits());
    println!("x32 {:08x}", x32.to_bits());
    println!("x32 as x64 {:016x}", (x32 as f64).to_bits());
}
```

I expected to see this happen:

No clippy warnings.

Instead, this happened: 

`4_294_967_296_f32` get reported as not being covered by f32. The output shows that this value could successfully be converted into the expected bit-pattern.
```
x64 41f0000000000000
x32 4f800000
x32 as x64 41f0000000000000
```

### Meta

- `cargo clippy -V`: clippy 0.1.54 (e82b650 2021-06-20)
- `rustc -Vv`:
  ```
  rustc 1.55.0-nightly (e82b65026 2021-06-20)
  binary: rustc
  commit-hash: e82b65026dcadf0998e3abf9dabc677f63160de7
  commit-date: 2021-06-20
  host: x86_64-unknown-linux-gnu
  release: 1.55.0-nightly
  LLVM version: 12.0.1
  ```

**Comment by kawogi** [comment]
If I understand the code correctly, clippy relies on a string comparison. The value in question results in a string output of `4294967300` (instead of `4294967296`) even though the internal representation is correct.
